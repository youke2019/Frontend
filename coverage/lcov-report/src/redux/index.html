<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.05% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/84</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>36/72</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="69.7" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.7" class="pct medium">69.7%</td>
	<td data-value="33" class="abs medium">23/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file low" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="24.49" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="25" class="abs low">9/25</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="48" class="abs low">12/48</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Aug 01 2019 01:16:28 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
<script src="..\..\block-navigation.js"></script>
</body>
</html>
