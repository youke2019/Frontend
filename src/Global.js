import Storage from 'react-native-storage'
import AsyncStorage from '@react-native-community/async-storage'

let storage = new Storage({
    size: 1000,
    storageBackend: AsyncStorage,
    defaultExpires: 1000 * 60 * 60 * 24, //1 day
    enableCache: true,
    sync: {}
})

global.storage = storage
global.baseUrl = 'http://192.168.42.146:8080'