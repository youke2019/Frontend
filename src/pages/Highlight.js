import React from 'react'
import { Image, FlatList, Text, View, ScrollView, TouchableOpacity, StyleSheet } from 'react-native'
import { connect } from 'react-redux'
import { Divider } from 'react-native-elements'
import HighlightCard from '../components/HighlightCard'
import { getAllHighlight, getNumberHighlight } from '../utils/DataRequest'

class Highlight extends React.Component {
  state = {
    highlights: [],
    receivedAll: false,
    is_refreshing:false,
  }
  newHighlight = () => {
    this.props.navigation.navigate('NewHighlight', {
      user_id: this.props.user_info.id,
      callBack: this.refresh
    })
  }
  componentDidMount () {
    this.getMoreData(20)
  }

  refresh = () => {
    this.refreshData()
  }
  getMoreData = (size)=>{
    const {highlights,receivedAll} = this.state
    if(receivedAll) return;
    const currNum = highlights.length
    this.setState(({
      is_refreshing: true
    }))
    getNumberHighlight(currNum, size, this.props.user_info.id)
      .then(response =>{
        this.setState({
          highlights: highlights.concat(response.data),
          is_refreshing:false,
        })
        if(response.data.length < size)
          this.setState({receivedAll:true})
      })
      .catch(err =>{
        console.log(err)
        this.setState({ is_refreshing: false })
      })
  }
  getTenMoreData=()=>{
    this.getMoreData(10);
  }

  refreshData = () => {
    const {highlights} = this.state
    const currNum = highlights.length
    this.setState({
      is_refreshing: true
    })
    getNumberHighlight(0,currNum,this.props.user_info.id)
      .then((response) => {
        console.log(response)
        this.setState({
          highlights: response.data,
          is_refreshing: false,
          receivedAll:false,
        })}) //刷新后，允许新的数据加入
      .catch(err => {
        this.setState({ is_refreshing: false })
        console.log(err)
      })
  }
  _renderItem=({ item }) =>
    <View style={{ flexDirection: 'column' }}>
      <HighlightCard data={item} user_id={this.props.user_info.id} refresh={this.refresh}/>
      <Divider style={{ height: 5, backgroundColor: 'whitesmoke' }}/>
    </View>
  _keyExtractor=(item) => item.video_id.toString()
  render () {
    const { highlights, is_refreshing } = this.state
    console.log(highlights)
    return (
      <View style={styles.base_container}>
        <View style={styles.header}>
          <View style={styles.header_center}>
            <Image source={{ uri: 'planet' }} style={styles.header_center_img}/>
            <Text style={styles.header_text}>
              精彩瞬间
            </Text>
          </View>
          <TouchableOpacity
            style={styles.top_right_touchable}
            onPress={this.newHighlight}
          >
            <Image source={{ uri: 'picture_yellow' }} style={styles.camera}/>
          </TouchableOpacity>
        </View>
        <Divider style={{ backgroundColor: 'black' }}/>
        <FlatList
          style={{ height: '100%', width: '100%' }}
          keyboardShouldPersistTaps={'handled'}
          refreshing={is_refreshing}
          onEndReached={this.getTenMoreData}
          onEndReachedThreshold={0.8}
          onRefresh={this.refresh}
          data={highlights}
          renderItem={this._renderItem}
          keyExtractor={this._keyExtractor}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  base_container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderTopWidth: 35,
    borderTopColor: 'whitesmoke'
  },
  header: {
    width: '100%',
    height: 50,
    flexDirection: 'row',
    backgroundColor: 'whitesmoke',
    alignItems: 'center',
    justifyContent: 'flex-end'
  },
  header_text: {
    color: '#000000',
    fontSize: 25,
    fontFamily: '字魂95号-手刻宋'
  },
  header_center: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  header_center_img: {
    height: 35,
    width: 35,
    marginRight: 10
  },
  camera: {
    height: 35,
    width: 35
  },
  top_right_touchable: {
    marginLeft: 80,
    marginRight: 20
  },
  divider: {
    backgroundColor: 'blue'
  }
})
const mapStateToProps = (state) => ({
  user_info: state.user_info
})

export default connect(
  mapStateToProps
)(Highlight)