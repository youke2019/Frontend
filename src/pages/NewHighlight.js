import { View, TextInput, Image, Text, StyleSheet, TouchableOpacity, Alert } from 'react-native'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Divider } from 'react-native-elements'
import ImagePicker from 'react-native-image-crop-picker'
import { sendHighlightImg, sendNewHighlight } from '../utils/DataRequest'

class StackNavBar extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired
  }
  goBack = () => {
    this.props.navigation.goBack()
  }

  render () {
    const { backIconUri = 'navigate-return' } = this.props
    return (
      <View style={navStyles.container}>
        <TouchableOpacity
          style={{ flex: 0.7, height: 20 }}
          activeOpacity={0.3}
          onPress={this.goBack}>
          <Image
            style={navStyles.backImage}
            source={{ uri: backIconUri }}
          />
        </TouchableOpacity>
        <TouchableOpacity
          onPress={this.props.onSend}
          style={navStyles.subNav}
        >
          <Image source={{ uri: 'paperplane' }} style={{ width: 30, height: 30 }}/>
          <Text style={{ textAlignVertical: 'center' }}>发表 </Text>
        </TouchableOpacity>
      </View>
    )
  }
}

const navStyles = StyleSheet.create({
  container: {
    marginTop: 35,
    height: 40,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  backImage: {
    marginLeft: 20,
    width: 25,
    height: 25
  },
  subNav: {
    marginRight: 20,
    flexDirection: 'row',
    justifyContent: 'center'
  }
})

const megas = {
  maxPicNumber: 1
}
const options = {
  title: '选择图片添加',
  customButtons: [{ name: 'fb', title: 'Choose Photo from Facebook' }],
  storageOptions: {
    skipBackup: true,
    path: 'images'
  }
}
export default class NewHighlight extends Component {
  state = {
    content: '',
    avatarSources: [],
  }
  send = () => {
    if (this.state.content === '') return;
    let urls =[]
    this.state.avatarSources.map(source=>{
      sendHighlightImg(source).then(response => {
        urls.push(response.data)
        if(urls.length === this.state.avatarSources.length){
          sendNewHighlight({
            user_id: this.props.navigation.state.params.user_id,
            post_text: this.state.content,
            image_url: urls[0],
            video_type:'i',
          }).then(response => {
            console.log(response)
            this.props.navigation.state.params.callBack()
            this.props.navigation.goBack()
            console.log('send')
          }).catch(err => console.log(err))
        }
      }).catch(err => console.log(err))
    })

  }
  uploadPic = () => {
    console.log('upload')
    ImagePicker.openPicker({
      width: 300,
      height: 400,
    }).then(image => {
      const file = {
        uri: image.path,
        type: 'multipart/form-data',
        name: 'image.jpg'
      }
      let oldSources = this.state.avatarSources;
      oldSources.push(file);
      this.setState({
        avatarSources:oldSources,
      })
    }).catch(err => console.log(err))
  }
  deleteImg = (index) => {
    Alert.alert(
      '提示',
      '是否删去图片?',
      [{
        text: '取消',
        onPress: () => {},
        style: 'cancel'
      }, {
        text: '确定',
        onPress: () => {this.delete(index)}
      }],
      { cancelable: true }
    )
  }
  delete = (index) => {
    let newSource = this.state.avatarSources
    newSource.splice(index, 1)
    this.setState({
      avatarSources: newSource
    })
  }
  updateContent = (content) => {
    this.setState({
      content: content
    })
  }

  render () {
    console.log(this.state)
    return (
      <View style={styles.base_container}>
        <StackNavBar
          navigation={this.props.navigation}
          onSend={this.send}
        />
        <Divider style={{ height: 5, backgroundColor: 'whitesmoke' }}/>
        <View style={styles.container}>
          <View style={styles.text_area}>
            <TextInput
              autoFocus
              maxLength={200}
              multiline
              style={styles.text_input}
              placeholder={'今天的课如何...'}
              value={this.state.content}
              onChangeText={this.updateContent}
            />
          </View>
          <View style={styles.pics}>
            {
              this.state.avatarSources.map((source, index) => {
                return (
                  <TouchableOpacity
                    onLongPress={() => {
                      this.deleteImg(index)
                    }}
                    key={index}
                    style={styles.img_button}>
                    <Image source={{ uri: source.uri }} style={styles.user_img}/>
                  </TouchableOpacity>
                )
              })
            }
            {
              this.state.avatarSources.length < megas.maxPicNumber &&
              <TouchableOpacity
                onPress={this.uploadPic}
                style={styles.img_button}>
                <Image source={{ uri: 'camera_yellow' }} style={styles.img}/>
                <Text style={styles.img_text}> 照片 / 视频</Text>
              </TouchableOpacity>
            }
          </View>
        </View>
      </View>
    )
  }
}
//TODO :You will also need to add UsageDescription on iOS and some permissions on Android, refer to the Install doc. IMPORTANT!!!
const styles = StyleSheet.create({
  base_container: {},
  container: {
    flexDirection: 'column',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    paddingTop: 20,
    paddingHorizontal: 15
  },
  text_area: {
    minHeight: 200
  },
  text_input: {
    lineHeight: 30,
    fontSize: 20
  },
  img_button: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: '30%',
    height: 120,
    backgroundColor: 'whitesmoke',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  user_img: {
    width: '100%',
    height: '100%'
  },
  img: {
    width: 35,
    height: 30
  },
  img_text: {
    paddingTop: 10,
    fontSize: 18,
    letterSpacing: 1,
    color: 'grey'
  },
  pics: {
    justifyContent: 'flex-start',
    flexDirection: 'row',
    flexWrap: 'wrap'
  }
})