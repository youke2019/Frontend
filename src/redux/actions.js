export const UPDATE_USER_INFO = 'UPDATEUSERINFO'
export const LOAD_USER_INFO = 'LOADUSERINFO'

export const updateUserInfo = (data) => ({
    type:UPDATE_USER_INFO,
    data
})

export const loadUserInfo = (data) => ({
    type:LOAD_USER_INFO,
    data
})

