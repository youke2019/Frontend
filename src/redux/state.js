const initialState = {
    user_info: null,
    course_list: null,
    login_ready:false,
    sortlist:[],
}

export const mockState = {
    course_list: [
        {"course_id":"GA312","course_name":"花卉学","course_hours":48,"course_credits":3,"general":false,"general_type":"","course_deptname":"设计学院"},
        {"course_id":"GA407","course_name":"花卉应用学","course_hours":32,"course_credits":2,"general":false,"general_type":"","course_deptname":"设计学院"},
        {"course_id":"PA302","course_name":"花卉与草虫禽鸟","course_hours":48,"course_credits":3,"general":false,"general_type":"","course_deptname":"媒体与传播学院"},
        {"course_id":"PL004","course_name":"插花艺术","course_hours":16,"course_credits":1,"general":false,"general_type":"","course_deptname":"设计学院"},
        {"course_id":"PL021","course_name":"花卉艺术","course_hours":32,"course_credits":2,"general":false,"general_type":"","course_deptname":"农业与生物学院"},
        {"course_id":"PL902","course_name":"国花、市花与文化","course_hours":32,"course_credits":2,"general":true,"general_type":"人文学科","course_deptname":"设计学院"},
        {"course_id":"SP079","course_name":"国花、市花鉴赏","course_hours":32,"course_credits":2,"general":false,"general_type":"","course_deptname":"设计学院"}
        ]
}

export default initialState