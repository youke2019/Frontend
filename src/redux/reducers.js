import { combineReducers } from 'redux'
import {
    UPDATE_USER_INFO,
    LOAD_USER_INFO,
    CLEAR_USER_INFO,
    SEARCH_COURSES,
} from "./actions.js"
import initialState from "./state";
import axios from 'axios'


function userInfoReducer(state=initialState.user_info, action) {
    switch (action.type) {
        case UPDATE_USER_INFO:
            storage.save({
                key: 'user',
                data: action.data
            })
            state = action.data
            return state
        case LOAD_USER_INFO:
            state = action.data
            return state
        case CLEAR_USER_INFO:
            state = null
            return state
        default:
            return state
    }
}

function courseReducer(state=initialState.course_list, action) {
    switch (action.type) {
        case SEARCH_COURSES:
            state = action.data
            return state
        default:
            return state
    }
}

export const combinedReducer = combineReducers({
    user_info : userInfoReducer,
    course_list: courseReducer,
})