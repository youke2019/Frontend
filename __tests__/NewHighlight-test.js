
import renderer from 'react-test-renderer';
import Global from '../src/Global'
import { createStore } from 'redux'
import { combinedReducer } from '../src/redux/reducers'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16/build'
import Highlight from '../src/pages/Highlight'
import React from 'react'
import NewHighlight from '../src/pages/NewHighlight'
import { isNullableTypeAnnotation } from '@babel/types'

const navigationMock = {
  navigate: jest.fn(()=>{}),
  callBack: jest.fn(()=>{}),
  user_id:"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A",
}
const initialState = {
  user_info: {
    id: "30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A",
  },
  course_list: null,
  login_ready:false,
}
const store = createStore(combinedReducer,initialState)
configure({adapter: new Adapter()})



jest.mock("../src/utils/DataRequest.js")

const refresh = jest.fn(()=>{})
const data = JSON.parse("{\"video_id\":1,\"user_id\":\"46420\",\"post_time\":\"177IG770\",\"post_text\":\"UC4GR662312XE4H5MBDU923644WLL3QE563EEQ7QBBT7\",\"video_url\":\"58850\",\"video_type\":\"W\",\"image_url\":\"3\",\"isbanned\":false,\"video_praise_point\":1,\"current_user_praise\":true,\"courseMomentCommentList\":[{\"video_comment_id\":2,\"video_id\":1,\"user_id\":\"14963\",\"video_comment_content\":\"07335\",\"video_comment_time\":\"69660\",\"isbanned\":true},{\"video_comment_id\":3,\"video_id\":1,\"user_id\":\"14963\",\"video_comment_content\":\"16292\",\"video_comment_time\":\"71693\",\"isbanned\":true},{\"video_comment_id\":6,\"video_id\":1,\"user_id\":\"34992\",\"video_comment_content\":\"74875\",\"video_comment_time\":\"95459\",\"isbanned\":true},{\"video_comment_id\":13,\"video_id\":1,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_comment_content\":\"还是是你的\",\"video_comment_time\":\"2019-07-20 23:22:26\",\"isbanned\":false},{\"video_comment_id\":14,\"video_id\":1,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_comment_content\":\"好的好的姐姐的\",\"video_comment_time\":\"2019-07-20 23:38:58\",\"isbanned\":false},{\"video_comment_id\":15,\"video_id\":1,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_comment_content\":\"防脱发头发\",\"video_comment_time\":\"2019-07-20 23:40:46\",\"isbanned\":false},{\"video_comment_id\":16,\"video_id\":1,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_comment_content\":\"刚才填充图册填充图册太吓人饭堂吃同学人吃肉馅肉馅长途车如此\",\"video_comment_time\":\"2019-07-20 23:42:05\",\"isbanned\":false},{\"video_comment_id\":18,\"video_id\":1,\"user_id\":\"1A7A1F80-59D0-4E8C-B4C6-AAAFFC606120\",\"video_comment_content\":\"你是真的牛逼\",\"video_comment_time\":\"2019-07-22 11:32:41\",\"isbanned\":false},{\"video_comment_id\":20,\"video_id\":1,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_comment_content\":\"？\",\"video_comment_time\":\"2019-07-22 16:54:50\",\"isbanned\":false}],\"courseMomentPraiseList\":[{\"video_praise_id\":11,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":12,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":13,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":14,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":15,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":16,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":17,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":18,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":19,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":20,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":21,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":22,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":23,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":24,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":25,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1},{\"video_praise_id\":26,\"user_id\":\"30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A\",\"video_id\":1}]}")
const user_id = "30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A";

test('renders newHighlight Page correctly', (done) => {
  const tree = renderer.create(
    <NewHighlight store = {store} navigation={navigationMock}/>
  ).toJSON();
  setTimeout(()=>{
    expect(tree).toMatchSnapshot();
    done();
  },1000)
});
describe("<NewHighlight /> component",()=>{
  const wrapper = shallow(
    <NewHighlight store = {store} navigation={navigationMock}/>
  );
  const instance = wrapper.instance();
  console.log(instance)
  it("will send request when click send icon",(done)=>{
    instance.setState({
      content:"123",
      avatarSources: [
        { source:1 },
      ]
    })
    instance.send();
    setTimeout(()=>{done()},2000)
    instance.setState({
      content:"123",
      avatarSources: [
        { source:1 },
      ]
    })
    instance.send();
    setTimeout(()=>{done()},2000)
  })
  it("will fail some times",(done)=>{
    instance.setState({
      content:"123",
      avatarSources: [
        { source:0 },
      ]
    })
    instance.send();
    setTimeout(()=>{done()},2000)
  })
  it("will update content when input text",()=>{
    instance.updateContent("1241234123");
  })
  it("will delete imgae when long press",()=>{
    instance.setState({
      avatarSources:[{no:1},{no:2}],
    })
    instance.delete(0);
  })
  it("will saveUploadImg to State when upload Img",()=>{
    instance.saveUploadFile({
      path:"path://faefaeffaesfaefefe"
    })
  })

})