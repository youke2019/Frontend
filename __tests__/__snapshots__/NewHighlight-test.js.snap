// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders newHighlight Page correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "marginTop": 35,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 0.7,
          "height": 20,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "navigate-return",
          }
        }
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "paperplane",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "textAlignVertical": "center",
          }
        }
      >
        发表 
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "whitesmoke",
        "height": 5,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "paddingHorizontal": 15,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "minHeight": 200,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        maxLength={200}
        multiline={true}
        onChangeText={[Function]}
        placeholder="今天的课如何..."
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 20,
            "lineHeight": 30,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "whitesmoke",
            "flexDirection": "column",
            "height": 120,
            "justifyContent": "center",
            "marginHorizontal": 5,
            "marginVertical": 10,
            "opacity": 1,
            "width": "30%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "camera_yellow",
            }
          }
          style={
            Object {
              "height": 30,
              "width": 35,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "letterSpacing": 1,
              "paddingTop": 10,
            }
          }
        >
           照片 / 视频
        </Text>
      </View>
    </View>
  </View>
</View>
`;
